framework/include/mesh/MooseMesh.h

```diff
@@ -418,11 +418,34 @@ class MooseMesh : public MooseObject, public Restartable, public PerfGraphInterf
   ///@}
 
   /**
-   * Returns a map of boundaries to elements.
+   * Returns a map of boundaries to ids of elements on the boundary.
    */
   const std::unordered_map<boundary_id_type, std::unordered_set<dof_id_type>> &
   getBoundariesToElems() const;
 
+  /**
+   * Returns a map of boundaries to ids of elements on the boundary.
+   */
+  const std::unordered_map<boundary_id_type, std::unordered_set<dof_id_type>> &
+  getBoundariesToActiveLocalElemIds() const;
+
+  /**
+   * Return all ids of elements which have a side which is part of a boundary/sideset.
+   * Note that boundaries are sided.
+   * @param bid the id of the boundary/sideset of interest
+   */
+  const std::unordered_set<dof_id_type> getBoundaryActiveLocalElemIds(BoundaryID bid) const;
+
+  /**
+   * Return all ids of neighbors of elements which have a side which is part of a boundary/sideset.
+   * Note that boundaries are sided, this is on the neighbor side. For the sideset side, use
+   * getBoundariesActiveLocalElemIds.
+   * Note that while the element is local and active, the neighbor is not guaranteed to be local,
+   * it could be ghosted.
```

Comment:
```suggestion
   * Note that while the element is local and active, the neighbor is not guaranteed to be local or active
```

Reply:
what happens if you don't have any side neighbor layers ghosted like for CFEM and you call this method when there are non-local neighbors?

Reply:
you should not get them with this code, because you ll retrieve some remote_elems and we exclude them

Reply:
:+1: Might want to mention that in the doxygen

Reply:
done